src/App.js

```diff
@@ -12,138 +12,101 @@ import People from './components/People';
 import Admin from './components/Admin';
 
-
 if (store.get("saved") === null) {
     store.set("saved", true);
 }
 
-const App = React.createClass({
-    loadPeopleFromServer: function(callback) {
-        $.ajax({
-            url: this.props.people_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.people, status, err.toString());
-            }
-        });
-    },
-    loadNicetiesFromMe: function(callback) {
-        $.ajax({
-            url: this.props.from_me_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.fromMe, status, err.toString());
-            }
-        });
-    },
-    loadNicetiesForMe: function(callback) {
-        $.ajax({
-            url: this.props.for_me_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.niceties, status, err.toString());
-            }
-        });
-    },
-    loadSelfInfo: function(callback) {
-        $.ajax({
-            url: this.props.self_api,
-            dataType: 'json',
-            cache: false,
-            success: function(data) {
-                callback(data);
-            },
-            error: function(xhr, status, err) {
-                //console.error(this.props.niceties, status, err.toString());
-            }
-        });
-    },
-    getInitialState: function() {
-        return {
-                fromMe: [],
-                people: [],
-                niceties: [],
-                currentview: "write-niceties",
-                selfInfo: [],
-        };
-    },
-    componentDidMount: function() {
-      this.loadNicetiesFromMe((data) => {
-        this.setState({fromMe: data})
-      })
-      this.loadPeopleFromServer((data) => {
-        this.setState({people: data})
-      })
-      this.loadNicetiesForMe((data) => {
-        this.setState({niceties: data})
+class App extends React.Component {
+    constructor(props) {
+      super(props)
+      this.state = {
+        fromMe: [],
+        people: [],
+        niceties: [],
+        currentview: "write-niceties",
+        selfInfo: []
+      }
+    }
+
+    loadData = (api, dataVar) => {
+      fetch(api, {
+        headers: {'Content-Type': "application/json"},
+        cache: 'no-cache'
       })
-      this.loadSelfInfo((data) => {
-        this.setState({selfInfo: data})
+      .then(response => response.json())
+      .then(data => {
+        let newState = {}
+        newState[dataVar] = data
+        this.setState(newState)
```

Comment:
originally this function took a callback, of the oneline form `(data) => this.setState({dataVar: data})`. I wanted to simplify loadData to take only the dataVar, which required this multiline object creation syntax. Is there a better way to do it?

I tried first calling e.g. `loadData(this.props.people_api, people)` and got a complaint about undefined variables, which is fair. Then I tried `loadData(this.props.people_api, this.state.people)` which gave no errors but also didn't work. Finally I settled on `loadData(this.props.people_api, "people")` which works when I construct the object as above, but not when I try the oneline `this.setState({dataVar: data})`.

Reply:
Ah, yeah this is a quirk of JavaScript object literals: because it allows for the syntax `{ literalString: variableName }`, you need to indicate that the key name is a variable: `{ [propertyName]: variableName }`. These are called [computed property names](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer#computed_property_names).

Altogether, then, you should be able to do:

```js
.then(data => this.setState({ [dataVar]: data })
```